.github/workflows/check.yml

```diff
@@ -20,6 +20,22 @@ jobs:
     if: github.event_name != 'push'
     name: Audit Dependencies 🕵️‍♂️
     uses: insightsengineering/r.pkg.template/.github/workflows/audit.yaml@main
+  r-cmd:
+    name: R CMD Check 🧬
+    uses: insightsengineering/r.pkg.template/.github/workflows/build-check-install.yaml@main
+    secrets:
+      REPO_GITHUB_TOKEN: ${{ secrets.REPO_GITHUB_TOKEN }}
+    with:
+      additional-env-vars: |
+        _R_CHECK_CRAN_INCOMING_REMOTE_=false
+      additional-r-cmd-check-params: --as-cran
+      enforce-note-blocklist: true
+      note-blocklist: |
+        checking dependencies in R code ... NOTE
+        checking R code for possible problems ... NOTE
+        checking examples ... NOTE
+        checking Rd line widths ... NOTE
+    # need a new path parameter set to "./package"
```

Comment:
FYI @cicdguy I have added r-cmd-check workflow but we are obviously miss path parameter. From what I noticed it's needed also for version-bump and version-check. It's nice to have for spelling but here we are not going to use it.

Reply:
The feature to specify a subdirectory in the repository, where the package is located, has just been merged to `r.pkg.template`. The new input called `package-subdirectory` is available in workflows in which the location of the package within the repository is relevant.

Reply:
Yes please. Thanks!

Reply:
Please have a look at 37cc362a3d6cc6a7f29b0d69caeae437d725bdfd - in particular: removed workflows to be added back as well as changed render call which might be very useful when debugging (I expect R CMD CHECK to take a while since we are rendering the whole book)

Reply:
I think the R CMD check is getting skipped because the [package name in `DESCRIPTION`](https://github.com/insightsengineering/tlg-catalog/actions/runs/5508960962/jobs/10040983744#step:18:25) is different than the [repository name](https://github.com/insightsengineering/tlg-catalog/actions/runs/5508960962/jobs/10040983744#step:20:49).

Reply:
> I think the R CMD check is getting skipped because the [package name in `DESCRIPTION`](https://github.com/insightsengineering/tlg-catalog/actions/runs/5508960962/jobs/10040983744#step:18:25) is different than the [repository name](https://github.com/insightsengineering/tlg-catalog/actions/runs/5508960962/jobs/10040983744#step:20:49).

This is quite uncommon but still it should be allowed. The most popular example is https://github.com/vubiostat/r-yaml that has `yaml` package inside. IMO: build step should return a path to .tar.gz file for check step to be picked up.

Reply:
I have changed package name to `tlg-catalog` to mach repo name but then R CMD BUILD failed as follows:
```
── R CMD build ──────────────────────────────────────────
   checking for file ‘/Users/ruckip/Documents/repo/gh/ins✔  checking for file ‘/Users/ruckip/Documents/repo/gh/insightsengineering/tlg-catalog/package/DESCRIPTION’
─  preparing ‘tlg-catalog’: (922ms)
E  checking DESCRIPTION meta-information ...
   Malformed package name
   
   See section 'The DESCRIPTION file' in the 'Writing R Extensions'
   manual.
Error in `(function (command = NULL, args = character(), error_on_status = TRUE, …`:
! System command 'R' failed
```

This workaround is not possible.

Reply:
Ok, I added the support to R CMD check workflow for packages with names different than the repository name.
Now the workflow run looks like [this](https://github.com/insightsengineering/tlg-catalog/actions/runs/5520600244/jobs/10067455146).

Reply:
actually the `staged_dependencies.yaml` file is at the repo root level so this should be picked up...

Reply:
ah no... the staged-deps step is actually a part of check workflow so it looks for that config file in the package subdir. let me move it

Reply:
I think we encountered this issue: https://github.com/openpharma/staged.dependencies/issues/172 🙂

Reply:
I don't think so because it's already set by [this input](https://github.com/insightsengineering/tlg-catalog/actions/runs/5520740988/jobs/10067799927#step:15:3) via [staged-dependencies-action](https://github.com/insightsengineering/staged-dependencies-action/blob/main/action.yml#L92).